### PR TITLE
WIP: Add access-logs.viewer.yaml

### PR DESCRIPTION
**Pull request by BobyMCbobs**
Add role for access-logs viewing.

Related: https://github.com/kubernetes/k8s.io/pull/1871

### CODE REVIEW

File: infra/gcp/roles/specs/access-logs.viewer.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+title: Access Logs
+description: View access to GCP access logs
+name: access-logs.viewer
+include:
+  roles:
+  - roles/compute.viewer
+  - roles/dns.reader
+  - roles/iam.securityReviewer
+  - roles/resourcemanager.organizationViewer
+  - roles/serviceusage.serviceUsageConsumer
```
Comment (spiffxp) [inline infra/gcp/roles/specs/access-logs.viewer.yaml:10]: Why are these roles necessary?

File: infra/gcp/roles/access-logs.viewer.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#### generated by generate-role-yaml.sh from specs/access-logs.viewer.yaml
```
Comment (spiffxp) [inline infra/gcp/roles/access-logs.viewer.yaml:1]: I don't believe this comment.  When I ran `./roles/generate-role-yaml.sh` I saw significantly more permissions show up

**Review by spiffxp — CHANGES_REQUESTED** [review]
Honestly I'm not even sure a custom role is necessary.

I think we can get by fine with stock roles on a GCS bucket, it will likely be a matter of pruning away things like `projectViewer` bindings

**Comment by spiffxp** [comment]
I put my thoughts on how I think we could setup IAM for gcs access logs here: https://github.com/kubernetes/k8s.io/issues/904#issuecomment-816484483

File: infra/gcp/roles/access-logs.viewer.yaml (hunk 2 above)
Comment (BobyMCbobs) [reply infra/gcp/roles/access-logs.viewer.yaml:1]: That's odd. I had to modify the script to get it to work locally
```diff
diff --git a/infra/gcp/roles/generate-role-yaml.sh b/infra/gcp/roles/generate-role-yaml.sh
index 7f5c8aae..4eba7280 100755
--- a/infra/gcp/roles/generate-role-yaml.sh
+++ b/infra/gcp/roles/generate-role-yaml.sh
@@ -68,13 +68,13 @@ function output_role_yaml() {
   local spec="${1}"
 
   local title description name include_roles include_permissions include_regex exclude_regex
-  title=$(<"${spec}" yq -r .title)
-  description=$(<"${spec}" yq -r .description)
-  name=$(<"${spec}" yq -r .name)
-  mapfile -t include_roles < <(<"${spec}" yq -r '.include? | .roles//[] | .[]')
-  mapfile -t include_permissions < <(<"${spec}" yq -r '.include? | .permissions//[] | .[]')
-  include_regex=$(<"${spec}" yq -r '.include? | .permissionRegexes//[] | join("|")')
-  exclude_regex=$(<"${spec}" yq -r '.exclude? | .permissionRegexes//[] | join("|")')
+  title=$(<"${spec}" yq e -M .title -)
+  description=$(<"${spec}" yq e -M .description -)
+  name=$(<"${spec}" yq e -M .name -)
+  mapfile -t include_roles < <(<"${spec}" yq e -M '.include? | .roles//[] | .[]' -)
+  mapfile -t include_permissions < <(<"${spec}" yq e -M '.include? | .permissions//[] | .[]' -)
+  include_regex=$(<"${spec}" yq e -M '.include? | .permissionRegexes//[] | join("|")' -)
+  exclude_regex=$(<"${spec}" yq e -M '.exclude? | .permissionRegexes//[] | join("|")' -)
 
   local output_path="${output_dir}/${name}.yaml"
```

I'm using [mikefarah/yq](https://github.com/mikefarah/yq), not [Python yq](https://pypi.org/project/yq/). To reproduce, the last commit before mine is 5de37c3784916ec2115adfbb3513519cd317fb4a. I think I must've broken something

File: infra/gcp/roles/specs/access-logs.viewer.yaml (hunk 1 above)
Comment (BobyMCbobs) [reply infra/gcp/roles/specs/access-logs.viewer.yaml:10]: I'm not particularly sure which roles may be needed, but would like to start the conversation

**Comment by ameukam** [comment]
@BobyMCbobs With https://github.com/kubernetes/k8s.io/pull/1966 merged, do we still need this?

**Comment by hh** [comment]
We have log access now. I think it's fine to close.
/close

**Pull request closed without merging**